src/main/java/org/folio/linked/data/mapper/kafka/inventory/KafkaInventoryMessageMapperImpl.java

```diff
@@ -20,8 +21,8 @@ public class KafkaInventoryMessageMapperImpl implements KafkaInventoryMessageMap
   private final ResourceModelMapper resourceModelMapper;
 
   @Override
-  public Optional<InstanceIngressPayload> toInstanceIngressPayload(Resource instance) {
-    if (isNull(instance) || isNull(instance.getId())) {
+  public Optional<InstanceIngressEvent> toInstanceIngressEvent(Resource instance) {
+    if (isNull(instance) || isNull(instance.getId()) || instance.isNotOfType(INSTANCE)) {
```

Comment:
This validation must be in service logic classes like senders, not in the mapper

Reply:
I agree, but we decided to do mapper refactoring in separated ticket, remember? :)

Reply:
discussed, agreed, changed